packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, Endian } from "../util/ByteArray.js";
+import { DataReader } from "../util/DataReader.js";
+import { DataWriter } from "../util/DataWriter.js";
+
+export interface BtpHandshakeRequest {
+    versions: number[],
+    attMtu: number,
+    clientWindowSize: number,
+}
+
+export interface BtpHandshakeResponse {
+    version: number,
+    attMtu: number,
+    windowSize: number
+}
+
+export interface BtpPacketPayload {
+    ackNumber?: number,
+    sequenceNumber: number,
+    msgLength?: number,
+    segmentPayload?: ByteArray
+}
+
+export interface BtpHeader {
+    handshakeBit: number,
```

Comment:
Document what these bits contain: `0 and 1` or `0 and MASK`.
If this is meant to be an API representation , possibly make them boolean?

Reply:
Yes these should be booleans in my eyes too

Reply:
Reverted it back to boolean